hie-plugin-api/Haskell/Ide/Engine/PluginUtils.hs

```diff
@@ -134,25 +144,70 @@ mapEithers _ _ = Right []
 
 -- ---------------------------------------------------------------------
 
--- |Generate a 'WorkspaceEdit' value from a pair of source Text
-diffText :: (Uri,T.Text) -> T.Text -> WorkspaceEdit
-diffText (f,fText) f2Text = WorkspaceEdit (Just h) Nothing
+data WithDeletions = IncludeDeletions | SkipDeletions
+  deriving Eq
+
+-- | Generate a 'WorkspaceEdit' value from an original file and text to replace it.
+makeDiffResult :: FilePath -> T.Text -> (FilePath -> FilePath) -> IdeM WorkspaceEdit
```

Comment:
Should this functionality move to haskell-lsp, some time?

Reply:
Probably, I imagine this would be useful for both clients and servers.

I'm also thinking we should probably also pass back the lsp funcs/lsp config back to haskell-lsp, so it can handle formatting the workspace edit according to the specification.